js/wp-seo-wordpress-scraper.js

```diff
@@ -0,0 +1,394 @@
+/**
+ * wordpress scraper to gather inputfields.
+ * @constructor
+ */
+YoastSEO_WordPressScraper = function(args, refObj) {
+    this.config = args;
+    this.refObj = refObj;
+    this.analyzerData = {};
+    this.formattedData = {};
+    this.formattedData.usedKeywords = wpseoMetaboxL10n.keyword_usage;
+	this.formattedData.searchUrl = "<a target='new' href="+wpseoMetaboxL10n.search_url+">";
+	this.formattedData.postUrl = "<a target='new' href="+wpseoMetaboxL10n.post_edit_url+">";
+    this.formattedData.queue = ["wordCount",
+        "keywordDensity",
+        "subHeadings",
+        "stopwords",
+        "fleschReading",
+        "linkCount",
+        "imageCount",
+        "urlKeyword",
+        "urlLength",
+        "metaDescription",
+        "pageTitleKeyword",
+        "pageTitleLength",
+        "firstParagraph",
+        "keywordDoubles"];
+    this.replacedVars = {};
+    this.getData();
+};
+
+/**
+ * Get data from inputfields and store them in an analyzerData object. This object will be used to fill
+ * the analyzer and the snippetpreview
+ */
+YoastSEO_WordPressScraper.prototype.getData = function() {
+    this.analyzerData.keyword = this.getDataFromInput( "keyword" );
+    this.analyzerData.meta = this.getDataFromInput( "meta" );
+    this.analyzerData.text = this.getDataFromInput( "text" );
+    this.analyzerData.pageTitle = this.getDataFromInput( "pageTitle" );
+    this.analyzerData.title = this.getDataFromInput( "title" );
+    this.analyzerData.url = this.getDataFromInput( "url" );
+    this.analyzerData.excerpt = this.getDataFromInput( "excerpt" );
+    this.analyzerData.snippetTitle = this.getDataFromInput( "snippetTitle" );
+    this.analyzerData.snippetMeta = this.getDataFromInput( "meta" );
+    this.analyzerData.snippetCite = this.getDataFromInput( "cite" );
+};
+
+/**
+ * gets the values from the given input. Returns this value
+ * @param inputType
+ * @returns value
+ */
+YoastSEO_WordPressScraper.prototype.getDataFromInput = function( inputType ) {
```

Comment:
Moreover, even though it might be more lines of code we should probably have 1 function per input type. This aids with autocomplete and makes the code much more readable and better to reason about.

Like this:
```js
function valueForId( id ) {
  return document.getElementById( id ).value;
}

function getPageTitle() {
  return valueForId( 'title' );
}

function getExcerpt() {
  return valueForId( 'excerpt' );
}
```